corehq/apps/hqwebapp/static/hqwebapp/spec/urllib_spec.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-env mocha */
+/* global $, sinon */
+
+describe('urllib', function() {
+    var urllib = hqImport('hqwebapp/js/urllib.js');
+    describe('getUrlParameterFromString', function() {
+        it('should return null when URL param missing', function() {
+            assert.equal(urllib.getUrlParameterFromString('asdf', '?limit=29'), null);
```

Comment:
Does this test pass? It looks like `getUrlParameterFromString` returns `undefined`, not `null`, for a missing parameter.